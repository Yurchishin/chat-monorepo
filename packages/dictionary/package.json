{
  "name": "@monorepo/dictionary",
  "version": "0.0.1",
  "description": "",
  "keywords": [],
  "author": "Yurii Rybak <yurarary@gmail.com>",
  "license": "ISC",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "watch": "tsc -p tsconfig.json --watch",
    "start": "node dist/index.js",
    "start:build": "npm run build & npm run start",
    "lint": "npm run lint:eslint npm run lint:tslint",
    "lint:fix": "npm run lint:eslint-fix npm run lint:tslint-fix",
    "lint:eslint": "eslint --ext .ts src",
    "lint:eslint-fix": "eslint --fix --ext .ts src",
    "lint:tslint": "tslint -c ./tslint.json -p ./tsconfig.json",
    "lint:tslint-fix": "tslint -c ./tslint.json -p ./tsconfig.json --fix"
  },
  "devDependencies": {
    "typescript": "^3.7.5"
  },
  "dependencies": {
    "@types/node": "^13.1.8"
  }
}
