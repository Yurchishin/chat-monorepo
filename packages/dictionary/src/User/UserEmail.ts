export const UserEmailError = {
    TYPE: 'Wrong email type',
    REG_EXP: 'Wrong email',
}
