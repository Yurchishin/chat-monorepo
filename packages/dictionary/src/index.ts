export * from './Hosts'
export * from './NodeEnv'
export const Hui = 'hi'
