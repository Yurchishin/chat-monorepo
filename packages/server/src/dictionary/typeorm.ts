export const TYPEORM_CONNECTION = 'postgres'
export const TYPEORM_ENTITIES = 'dist/entity/**/*.js'
export const TYPEORM_MIGRATIONS = 'dist/migration/**/*.js'
export const TYPEORM_ENTITIES_DIR = 'src/entity'
export const TYPEORM_MIGRATIONS_DIR = 'src/migration'

