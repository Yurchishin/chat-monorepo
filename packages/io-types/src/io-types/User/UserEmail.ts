import * as t from 'io-ts'
import * as E from 'fp-ts/lib/Either'
import isEmail from 'validator/lib/isEmail'
import { withMessage } from 'io-ts-types/lib/withMessage'
import { UserEmailError } from '@monorepo/dictionary'

export interface IUserEmailC extends t.Type<string> {}

export const userEmail: IUserEmailC = new t.Type<string>(
    'UserEmail',
    t.string.is,
    (u, c) =>
        E.either.chain(
            withMessage(t.string, () => UserEmailError.TYPE)
                .validate(u, c),
            s => isEmail(s) ? t.success(s) : t.failure(u, c, UserEmailError.REG_EXP),
        ),
    t.identity,
)

export type TUserEmail = t.TypeOf<typeof userEmail>
