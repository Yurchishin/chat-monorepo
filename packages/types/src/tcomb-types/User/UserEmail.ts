import t, { Intersection } from 'tcomb-validation'
import { UserEmailError } from '@monorepo/dictionary'
import { addErrorMessage } from '../../utils'
import tString from '../String'

export const UserEmail: Intersection<string> = t.intersection([
    addErrorMessage(tString.String, UserEmailError.TYPE),
    addErrorMessage(tString.Email, UserEmailError.REG_EXP),
], 'UserEmail')
