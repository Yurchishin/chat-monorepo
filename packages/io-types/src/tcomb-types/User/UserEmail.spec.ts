import {UserEmail} from './UserEmail'
import {expectTypeSuccess, expectTypeErrors} from "../../tests"
import { UserEmailError } from "@monorepo/dictionary";

describe('UserEmail', () => {
	expectTypeSuccess(UserEmail, 'yurarary158@gmail.com')
	expectTypeErrors(UserEmail, 'yurarary', [UserEmailError.REG_EXP])
	expectTypeErrors(UserEmail, 'yurarary@', [UserEmailError.REG_EXP])
	expectTypeErrors(UserEmail, 'yurarary@12123', [UserEmailError.REG_EXP])
	expectTypeErrors(UserEmail, 1, [UserEmailError.TYPE, UserEmailError.TYPE])
})
