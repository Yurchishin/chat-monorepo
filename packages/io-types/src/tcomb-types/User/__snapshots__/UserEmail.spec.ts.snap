// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEmail expectTypeFailure(1): TYPE_FAILURE::UserEmail(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong email type",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong email type",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`UserEmail expectTypeFailure(yurarary): TYPE_FAILURE::UserEmail(yurarary) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "yurarary",
      "expected": [Function],
      "message": "Wrong email",
      "path": Array [],
    },
  ],
  "value": "yurarary",
}
`;

exports[`UserEmail expectTypeFailure(yurarary@): TYPE_FAILURE::UserEmail(yurarary@) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "yurarary@",
      "expected": [Function],
      "message": "Wrong email",
      "path": Array [],
    },
  ],
  "value": "yurarary@",
}
`;

exports[`UserEmail expectTypeFailure(yurarary@12123): TYPE_FAILURE::UserEmail(yurarary@12123) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "yurarary@12123",
      "expected": [Function],
      "message": "Wrong email",
      "path": Array [],
    },
  ],
  "value": "yurarary@12123",
}
`;
